Update Swift Test App for iOS 13 compatibility

timkimadobe opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

## Description

This PR updates the test app for iOS 13 compatibility; some newer SwiftUI additions in iOS 14 were rolled back to iOS 13 implementation. This means that the `@main` and `@StateObject` property wrappers are not available, and app is now initialized using the app + scene delegate flow, and `@StateObject` is reverted to `@ObservedObject`.

## Related Issue

<!--- This project only accepts pull requests related to open issues -->
<!--- If suggesting a new feature or change, please discuss it in an issue first -->
<!--- If fixing a bug, there should be an issue describing it with steps to reproduce -->
<!--- Please link to the issue here: -->

## Motivation and Context

Allows for testing new ad ID features in iOS 13, while still using SwiftUI for views.

## How Has This Been Tested?

Simulator running iOS 13.

## Types of changes

<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->

- [x] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

timkimadobe committed:
Update ad ID view to use ad ID utils and better align with sample app implementation

timkimadobe committed:
Update requestTrackingAuthorization function to be pure request tracking flow; other functionality is handled by other methods in the util class

timkimadobe committed:
Merge branch 'feature/adIdConsent' into feature/ad-id-consent

# Conflicts:
#	Podfile.lock
#	SampleApps/TestApp/ContentView.swift

timkimadobe committed:
Remove get ad ID in request flow; use get ad ID flow instead to fetch or print value

timkimadobe committed:
Update documentation wording to defer to apple docs

timkimadobe committed:
Update documentation to reflect iOS version verbiage

timkimadobe committed:
Updating podfile.lock to use staging

timkimadobe committed:
Update logical flow for setting ad ID to check for tracking authorization and set the value accordingly

timkimadobe committed:
Update Swift test app to use app + scene delegate initialization for iOS 13 compatibility

timkimadobe committed:
Merge branch 'feature/adIdConsent' into feature/ad-id-consent

# Conflicts:
#	AEPEdgeIdentity.xcodeproj/project.pbxproj

emdobrin commented on SampleApps/TestApp/ContentView.swift:
```diff
@@ -258,40 +261,40 @@ struct CustomIdentifierView: View {
                         MobileCore.registerExtension(AEPEdgeIdentity.Identity.self)
                     }
-
+                    
```
Can you run `make format` or are these changes after formatting?

timkimadobe replied:
Thank you, forgot to run the autocorrect! Should be applied with the latest commit

emdobrin commented on SampleApps/TestApp/SceneDelegate.swift:
```diff
@@ -0,0 +1,33 @@
+    var window: UIWindow?
+    
+    func scene(_ scene: UIScene, willConnectTo session: UISceneSession, options connectionOptions: UIScene.ConnectionOptions) {
+//        let window = UIWindow(frame: UIScreen.main.bounds)
+//        window.rootViewController = UIHostingController(rootView: ContentView())
+//        self.window = window
+//        window.makeKeyAndVisible()
+        
+        // Use a UIHostingController as window root view controller.
+        if let windowScene = scene as? UIWindowScene {
```
let's remove commented code and simplify the if { } with a guard 

guard let windowScene = (scene as? UIWindowScene) else { return }
window = ...

timkimadobe replied:
Updated to guard statement!

emdobrin commented on SampleApps/TestApp/TestApp.swift:
```diff
@@ -11,15 +11,15 @@
-        }
-    }
-}
+// Implementation updated to use App + Scene Delegates for iOS 13 compatibility
```
let's remove this file if no longer used

timkimadobe replied:
Removed file

timkimadobe committed:
Update test app to include storyboard which is required to use full screen

timkimadobe committed:
Remove TestApp App SwiftUI initialization file (replaced with app + scene delegates)
Apply swiftlint autocorrect and update Button closure syntax to avoid trailing closure violation

timkimadobe committed:
Update UIWindowScene cast to guard statement

timkimadobe reviewed (commented):
Thanks for the review @emdobrin! Applied lint autocorrect and applied suggested changes

timkimadobe committed:
Update modifier positions for consistency, remove unnecessary whitespace
